[shelly] Fix initialization of BLU Motion device

markus7017 opened the pull request:
This PR fixes BLU motion initialization. In facts various flag in the device profile were not correctly initialized. This also affected other BLU devices even they seem to work.

This also should be back ported to 4.2 (cheeky picking)

Fixes https://github.com/openhab/openhab-addons/issues/17052

markus7017 committed:
Fixes #17052 (caused by inproper initialization of the profile flags for
BLU devices in general and for BLU Motion another one)

Signed-off-by: Markus Michels <markus7017@gmail.com>

lolodomo commented:
"awaiting feedback" tag is used when a reviewer is awaiting a feedback from the contributor.

jlaur commented on bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:
```diff
@@ -293,8 +293,11 @@ public boolean initializeThing() throws ShellyApiException {
-                getThing().getLabel(), thingType, config.deviceAddress, gen2, config.eventsCoIoT);
+        profile.initFromThingType(thingType);
+        logger.debug(
+                "{}: Start initializing for thing {},  type {}, Device address {}, Gen2: {}, isBlu: {}, alwaysOn: {}, hasBattery: {}, CoIoT: {}",
```
```suggestion
                "{}: Start initializing for thing {}, type {}, Device address {}, Gen2: {}, isBlu: {}, alwaysOn: {}, hasBattery: {}, CoIoT: {}",
```

jlaur replied:
Marked as resolved without comment or commit?

jlaur commented on bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:
```diff
@@ -293,8 +293,11 @@ public boolean initializeThing() throws ShellyApiException {
+        profile.initFromThingType(thingType);
+        logger.debug(
+                "{}: Start initializing for thing {},  type {}, Device address {}, Gen2: {}, isBlu: {}, alwaysOn: {}, hasBattery: {}, CoIoT: {}",
+                thingName, getThing().getLabel(), thingType, config.deviceAddress.toUpperCase(), gen2, profile.isBlu,
```
For a debug message, wouldn't it be better to have the actual value rather than an uppercased version which can be derived from the original anyway?
```suggestion
                thingName, getThing().getLabel(), thingType, config.deviceAddress, gen2, profile.isBlu,
```

markus7017 replied:
the value itself works in upper/lower-case, but I think uppercase makes HEX more readable

jlaur replied:
OK. My point was the it's a debug message, so it seemed logical to have as much information as possible, and by converting to uppercase you are losing information. But it's a minor comment, it's also fine as is.

jlaur reviewed (changes requested):
Thanks for the fix! I have added two very minor comments regarding logging, otherwise LGTM. Please ping when tested/verified (I replaced the label, see @lolodomo's comment).

jlaur commented on bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java:
```diff
@@ -236,8 +236,9 @@ public void initFromThingType(String name) {
-        hasBattery = isHT || isFlood || isDW || isSmoke || isButton || isMotion || isTRV;
-        alwaysOn = !hasBattery || isMotion || isSense; // true means: device is reachable all the time (no sleep mode)
+        hasBattery = isHT || isFlood || isDW || isSmoke || isButton || isMotion || isTRV || isBlu;
+        alwaysOn = !hasBattery || (isMotion && !isBlu) || isSense; // true means: device is reachable all the time (no
```
Just to be sure: A battery-powered non-Bluetooth motion sensor is considered always on?

markus7017 replied:
yes, it has kind of a low power mode; normale sensors and BLU devices are really in deep sleep and can't be waked up by an application

jlaur replied:
So a "normal" (non-Bluetooth) battery-powered motion sensor has a low power mode, but is still reachable? What do you mean by "normal sensors" then, from your answer this seems to exclude motion sensors?

markus7017 replied:
Shelly Motion is the only one with this low power mode, having a battery. Sensors like Flood, H&T as well as BLU devices goto into deep sleep and are not reachable

markus7017 commented:
@jlaur I did some testing with 10 BLU DW and 2 BLU Motion sensors, also 3 different Shelly acting as hub
I user confirmed the fix, solved this problem.

Nevertheless, I saw "BLU Device not yet connected" which shouldn't show up on device initialization. I'm still chasing, it's hard to reproduce so it makes sense to merge this PR and add another one when I have a fix.

jlaur committed:
Remove excess space

Signed-off-by: Jacob Laursen <jacob-github@vindvejr.dk>

jlaur commented on bundles/org.openhab.binding.shelly/README.md:
```diff
@@ -207,12 +213,16 @@ Follow these steps to add the Shelly BLU Device to openHAB
 
-Note: During initialization the script 'oh-blu-scanner.js' gets installed and activated on the Shelly Gateway device.
+`Note`:
+- During initialization the script 'oh-blu-scanner.js' gets installed and activated on the Shelly Gateway device.
```
```suggestion

- During initialization the script 'oh-blu-scanner.js' gets installed and activated on the Shelly Gateway device.
```

jlaur commented on bundles/org.openhab.binding.shelly/README.md:
```diff
@@ -207,12 +213,16 @@ Follow these steps to add the Shelly BLU Device to openHAB
+`Note`:
+- During initialization the script 'oh-blu-scanner.js' gets installed and activated on the Shelly Gateway device.
+- Shelly BLU Motion: It may take some time until channels like Lux show up.
+Try moving the device to force status updates.
```
```suggestion

Try moving the device to force status updates.
```

jlaur committed:
Apply suggestions from code review

Fix:
```
[WARNING] .binding.shelly/README.md:[217]
The line before a Markdown list must be empty.
[WARNING] .binding.shelly/README.md:[218]
The line after a Markdown list must be empty.
```

Signed-off-by: Jacob Laursen <jacob-github@vindvejr.dk>
